meta-facebook/aalc-rpu/src/platform/plat_modbus.c

```diff
@@ -0,0 +1,733 @@
+/*
+ * Copyright (c) Meta Platforms, Inc. and affiliates.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ * 
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+#include <stdlib.h>
+#include <sys/util.h>
+#include <sys/byteorder.h>
+#include <drivers/gpio.h>
+#include <modbus/modbus.h>
+#include <time.h>
+#include <logging/log.h>
+#include "sensor.h"
+#include "modbus_server.h"
+#include "fru.h"
+#include "eeprom.h"
+#include "libutil.h"
+#include "plat_modbus.h"
+#include "plat_sensor_table.h"
+#include "plat_control.h"
+#include "plat_fru.h"
+#include "hal_gpio.h"
+#include "plat_gpio.h"
+#include "plat_fw_update.h"
+#include <modbus_internal.h>
+#include "plat_util.h"
+#include "libutil.h"
+#include "plat_pwm.h"
+#include "util_sys.h"
+
+LOG_MODULE_REGISTER(plat_modbus);
+
+#define FW_UPDATE_SWITCH_FC 0x64
+#define FW_UPDATE_SWITCH_ADDR 0x0119
+#define FW_UPDATE_ENABLE_DATA 0x0101
+#define FW_UPDATE_DISABLE_DATA 0x0100
+
+//{ DT_PROP(DT_INST(0, zephyr_modbus_serial), label) }
+
+typedef struct {
+	char *iface_name;
+	bool is_custom_fc64;
+} modbus_server;
+modbus_server modbus_server_config[] = {
+	// iface_name, is_custom_fc64
+	{ "MODBUS0", true },
+	{ "MODBUS1", false },
+};
+
+static float pow_of_10(int8_t exp)
+{
+	float ret = 1.0;
+	int i;
+
+	if (exp < 0) {
+		for (i = 0; i > exp; i--) {
+			ret /= 10.0;
+		}
+	} else if (exp > 0) {
+		for (i = 0; i < exp; i++) {
+			ret *= 10.0;
+		}
+	}
+
+	return ret;
+}
+
+/*
+	arg0: sensor number
+	arg1: m
+	arg2: r
+
+	actual_val =  raw_val * m * (10 ^ r)
+*/
+
+static uint8_t modbus_get_senser_reading(modbus_command_mapping *cmd)
+{
+	CHECK_NULL_ARG_WITH_RETURN(cmd, MODBUS_EXC_ILLEGAL_DATA_VAL);
+
+	int reading = 0;
+	uint8_t status = get_sensor_reading(sensor_config, sensor_config_count, cmd->arg0, &reading,
+					    GET_FROM_CACHE);
+
+	if (status == SENSOR_READ_SUCCESS) {
+		sensor_val *sval = (sensor_val *)&reading;
+		float val = (sval->integer * 1000 + sval->fraction) / 1000;
+		float r = pow_of_10(cmd->arg2);
+		uint16_t byte_val = val / cmd->arg1 / r; // scale
+		memcpy(cmd->data, &byte_val, sizeof(uint16_t) * cmd->cmd_size);
+		return MODBUS_EXC_NONE;
+	}
+
+	return MODBUS_EXC_SERVER_DEVICE_FAILURE;
+}
+
+static uint8_t modbus_to_do(modbus_command_mapping *cmd)
```

Comment:
Can you please name this modbus_not_supported or modbus_not_implemented instead?

Reply:
Hi, I changed function name "modbus_to_do" to "modbus_not_implemented", thanks!